File: src/market-tracker-webapi/Application/Service/Operations/Market/Alert/IAlertService.cs
using market_tracker_webapi.Application.Domain.Schemas.Market.Retail.Sales.Pricing;

namespace market_tracker_webapi.Application.Service.Operations.Market.Alert;

public interface IAlertService
{
    Task<IEnumerable<PriceAlert>> GetPriceAlertsByClientIdAsync(Guid clientId, string? productId, int? storeId);

    Task<PriceAlertId> AddPriceAlertAsync(
        Guid clientId,
        string productId,
        int storeId,
        int priceThreshold
    );

    Task<PriceAlert> RemovePriceAlertAsync(Guid clientId, string alertId);
}

File: src/market-tracker-android/app/src/main/java/pt/isel/markettracker/http/service/operations/alert/IAlertService.kt
package pt.isel.markettracker.http.service.operations.alert

import pt.isel.markettracker.domain.model.market.price.PriceAlert
import pt.isel.markettracker.domain.model.market.price.PriceAlertId

interface IAlertService {
    suspend fun getAlerts(): List<PriceAlert>

    suspend fun createAlert(
        productId: String,
        storeId: Int,
        priceThreshold: Int
    ): PriceAlertId

    suspend fun deleteAlert(alertId: String)
}